src/datajudge/db_access.py

```diff
@@ -632,43 +632,65 @@ def get_row_count(engine, ref, row_limit: int = None):
     return result, [selection]
 
 
-def _column_aggregate(engine, ref, column_operator):
+def _column(engine, ref, aggregate_operator: Callable = None):
+    """
+    Queries the database for the values of the relevant column (as returned by `get_column(...)`).
+    If an aggregation operation is passed, the results are aggregated accordingly
+    and a single scalar value is returned.
+    @param engine: SQLAlchemy engine
+    @param ref: DataReference object
+    @param aggregate_operator: Aggregation operation, e.g. SQLAlchemy's `sa.func.min`
+    @return: Scalar value if aggregate_operator is requested, otherwise list of values in column
+    """
     subquery = ref.get_selection(engine).alias()
     column = subquery.c[ref.get_column(engine)]
-    selection = sa.select([column_operator(column)])
-    result = engine.connect().execute(selection).scalar()
+
+    if not aggregate_operator:
+        selection = sa.select([column])
+        result = engine.connect().execute(selection).all()
```

Comment:
Maybe it might make sense to use `.scalars()`instead of `.all()` here. :)
https://docs.sqlalchemy.org/en/14/core/connections.html?highlight=fetchall#sqlalchemy.engine.Result.scalars